simple_network_sim/loaders.py

```diff
@@ -179,6 +179,53 @@ def readInfectiousStates(infectious_states: pd.DataFrame) -> List[Compartment]:
     return list(infectious_states.Compartment)
 
 
+def readABCSMCParameters(parameters: pd.DataFrame) -> Dict:
+    """
+    Transforms the API output of parameters into the internal representation: a dict
+    :param parameters: pandas DataFrame with the raw data from the API
+    :return: a dict of inference parameters
+    """
+    if parameters.size == 0:
+        return {}
+
+    if "Parameter" not in parameters.columns:
+        raise ValueError("'Parameter' column should be in ABCSMC parameters")
+
+    if "Value" not in parameters.columns:
+        raise ValueError("'Value' column should be in ABCSMC parameters")
+
+    parameters = parameters.set_index("Parameter").Value.to_dict()
+
+    parameters["n_smc_steps"] = int(parameters["n_smc_steps"])
+    parameters["n_particles"] = int(parameters["n_particles"])
+    parameters["infection_probability_shape"] = float(parameters["infection_probability_shape"])
+    parameters["infection_probability_kernel_sigma"] = float(parameters["infection_probability_kernel_sigma"])
+    parameters["initial_infections_stddev"] = float(parameters["initial_infections_stddev"])
+    parameters["initial_infections_stddev_min"] = float(parameters["initial_infections_stddev_min"])
+    parameters["initial_infections_kernel_sigma"] = float(parameters["initial_infections_kernel_sigma"])
+
+    return parameters
+
+
+def readHistoricalDeaths(historical_deaths: pd.DataFrame) -> pd.DataFrame:
+    """
+    Transforms the API output of target into the internal representation: a pd.DataFrame
+    :param historical_deaths: pandas DataFrame with the raw data from the API
+    :return: a pd.DataFrame of historical deaths by HB
+    """
+    if historical_deaths.size == 0:
+        raise ValueError("With an empty target no inference can take place")
+
+    historical_deaths = historical_deaths.set_index("Week beginning")
+
+    if len(historical_deaths.columns) != 14:
+        raise ValueError("The historical deaths should contain a column for every HB")
```

Comment:
This number of columns is specific to health boards in Scotland, and we are likely to want to run inference on other geographies at some point.  If generalisation isn't an option at this point, then perhaps pointing out this HB-in-Scotland dependence a little more stridently in the comments is advisable.

Reply:
Agreed, I think I added this check when I wanted to make sure my data was fine, but we should remove it.